librosa/display.py

```diff
@@ -844,7 +844,7 @@ def cmap(
     data = np.atleast_1d(data)
 
     if data.dtype == "bool":
-        return mcm.get_cmap(cmap_bool, lut=2)
+        return mcm[cmap_bool]
```

Comment:
This one seems potentially wrong, since we're losing the explicit binary map implied by `lut=2`.  Can we still get that with the newer API?

Reply:
If ```matplotlib``` detects what is normally considered binary data, it will binarise the ```cmap``` as this is a very common usage pattern in computer vision:

```python
import matplotlib.pyplot as plt
import numpy as np

fig, ax = plt.subplots()
data = [[0, 1, 0, 1], [1, 0, 1, 0]]
ax.imshow(data, cmap="gray")
fig.savefig("test_int8_bool.png")

fig, ax = plt.subplots()
data = [[False, True, False, True], [True, False, True, False]]
ax.imshow(data, cmap="gray")
fig.savefig("test_bool.png")

fig, ax = plt.subplots()
data = [[0, 255, 0, 255], [255, 0, 255, 0]]
ax.imshow(data, cmap="gray")
fig.savefig("test_int8_max.png")

fig, ax = plt.subplots()
data = [[0.0, 1.0, 0.0, 1.0], [1.0, 0.0, 1.0, 0.0]]
ax.imshow(data, cmap="gray")
fig.savefig("test_float.png")
```

These will all produce:

![test_bool](https://github.com/librosa/librosa/assets/20606501/479719f9-1d4a-4f0e-8eab-0572769cb3ae)

Reply:
I would have to go digging through ```matplotlib``` issues and PR's to find the exact reason for removing the ```lut``` kwarg, but personally I would transform (e.g., binarise) my data prior to sending it to ```matplotlib``` to display.

Reply:
Ok, thanks for clarification!  It's entirely possible that our `lut=2` code predates matplotlib doing this automatically, so it might be entirely moot anyway.  I suppose if the image comparison tests are still passing after this change though, we should be fine.

Reply:
> but personally I would transform (e.g., binarise) my data prior to sending it to `matplotlib` to display.

Sorry, I missed this part earlier.  What we're actually doing here is selecting a colormap based on the data - if we see dtype=bool (ie it's already binary) then we use black-and-white (the code in question).  So I think we don't actually need to change anything here, if matplotlib is behaving as you say above and setting the number of colors appropriately.